index.js

```diff
@@ -65,15 +65,20 @@ function install (id) {
 /**
  * Uses webpack to build a file in memory and return the bundle.
  *
- * @param {*} file - the entry file to build.
+ * @param {*} name - library folder name
+ * @param {*} file - entry point path relative to the library folder
  * @return {Promise<string>}
  */
-function build (file) {
+function build (name, file) {
   return new Promise((resolve, reject) => {
+    const entry = requireRelative.resolve(path.join(name, file), tmp);
+    const packageJson = requireRelative.resolve(path.join(name, 'package.json'), tmp);
+    const externals = Object.keys(require(packageJson).peerDependencies || {});
```

Comment:
I wonder if it's possible to declare the same thing in both `peerDeps` and regular `deps`. If it is then we should exclude things that are in regular `deps` from here. But I'm not sure npm would allow this anyway. Worth a check.